compiler/src/dmd/traits.d

```diff
@@ -992,69 +992,90 @@ Expression semanticTraits(TraitsExp e, Scope* sc)
         e.ident == Id.getVirtualMethods ||
         e.ident == Id.getVirtualFunctions)
     {
-        if (dim != 2 && !(dim == 3 && e.ident == Id.getOverloads))
+        if (e.ident == Id.getOverloads)
+        {
+            if (dim != 1 && dim != 2 && dim != 3)
+            {
+                e.error("expected 1, 2 or 3 arguments for `__traits(getOverloads)` but had %d", cast(int)dim);
+                return ErrorExp.get();
+            }
+        }
+        else if (dim != 2)
             return dimError(2);
 
+        bool includeTemplates = false;
+        Expression ex = null;
+        Dsymbol sym = null;
```

Comment:
Why is `sym` declared here, when its first use is on line 1059?

Reply:
Because it's filled in the `else` branch for the `Id.getOverloads && dim == 1` test.